packages/schematics/angular/web-worker/index.ts

```diff
@@ -0,0 +1,172 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { JsonParseMode, parseJsonAst, strings, tags } from '@angular-devkit/core';
+import {
+  Rule, SchematicContext, SchematicsException, Tree,
+  apply, applyTemplates, chain, mergeWith, move, noop, url,
+} from '@angular-devkit/schematics';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import { getWorkspace, updateWorkspace } from '../utility/config';
+import { appendValueInAstArray, findPropertyInAstObject } from '../utility/json-utils';
+import { parseName } from '../utility/parse-name';
+import { buildDefaultPath, getProject } from '../utility/project';
+import { getProjectTargets } from '../utility/project-targets';
+import {
+  BrowserBuilderOptions,
+  BrowserBuilderTarget,
+  WorkspaceSchema,
+} from '../utility/workspace-models';
+import { Schema as WebWorkerOptions } from './schema';
+
+function getProjectConfiguration(
+  workspace: WorkspaceSchema,
+  options: WebWorkerOptions,
+): BrowserBuilderOptions {
+  const projectTargets = getProjectTargets(workspace, options.project);
+  if (!projectTargets[options.target]) {
+    throw new Error(`Target is not defined for this project.`);
+  }
+
+  const target = projectTargets[options.target] as BrowserBuilderTarget;
+
+  return target.options;
+}
+
+function addConfig(options: WebWorkerOptions, root: string): Rule {
+  return (host: Tree, context: SchematicContext) => {
+    context.logger.debug('updating project configuration.');
+    const workspace = getWorkspace(host);
+    const config = getProjectConfiguration(workspace, options);
+
+    if (!!config.experimentalWebWorkerTsConfig) {
+      // Don't do anything if the configuration is already there.
+      return;
+    }
+
+    config.experimentalWebWorkerTsConfig =
+      `${root.endsWith('/') ? root : root + '/'}tsconfig.worker.json`;
+
+    // Add project tsconfig.json and tsconfig.worker.json.
+    // The project level tsconfig.json with webworker lib is for editor support since
+    // the dom and webworker libs are mutually exclusive.
+    // Note: this schematic does not change other tsconfigs to use the project-level tsconfig.
```

Comment:
This schematic makes a two new tsconfig files:
- `path/to/project/tsconfig.worker.json` for the worker TS compilation itself
- `path/to/project/tsconfig.json` for the editor to support both DOM and WebWorker typings inside the project

As is I did not make the remaining tsconfigs in `path/to/project/` extend `path/to/project/tsconfig.json` because it's hard to ensure they are the default tsconfigs we generate on a new project.

But I believe we should eventually transition into having a `path/to/project/tsconfig.json` anyway in all projects because of other editor support issues like https://github.com/angular/angular-cli/issues/8138#issuecomment-458189246.